R/summariseRecordCount.R

```diff
@@ -28,41 +29,59 @@
 #')
 #'
 #'# Run summarise clinical tables
-#'summarisedResult <- summariseRecordCount(omopTable = cdm$condition_occurrence,
-#'                                       unit = "year",
-#'                                       unitInterval = 10,
-#'                                       ageGroup = list("<=20" = c(0,20), ">20" = c(21, Inf)),
-#'                                       sex = TRUE)
+#'summarisedResult <- summariseRecordCount(cdm = cdm,
+#'                                          omopTableName = "condition_occurrence",
+#'                                          unit = "year",
+#'                                          unitInterval = 10,
+#'                                          ageGroup = list("<=20" = c(0,20), ">20" = c(21, Inf)),
+#'                                          sex = TRUE)
 #'summarisedResult |> print()
 #'}
-summariseRecordCount <- function(omopTable, unit = "year", unitInterval = 1, ageGroup = NULL, sex = FALSE) {
+summariseRecordCount <- function(cdm, omopTableName, unit = "year",
+                                 unitInterval = 1, ageGroup = NULL, sex = FALSE) {
 
   # Initial checks ----
-  checkOmopTable(omopTable)
+  omopgenerics::validateCdmArgument(cdm)
+  omopgenerics::assertCharacter(omopTableName)
 
   if(missing(unit)){unit <- "year"}
   if(missing(unitInterval)){unitInterval <- 1}
   if(missing(ageGroup) | is.null(ageGroup)){ageGroup <- NULL}
 
   checkUnit(unit)
   checkUnitInterval(unitInterval)
-  checkAgeGroup(ageGroup)
-
-  assertLogical(sex, length = 1)
+  omopgenerics::validateAgeGroupArgument(ageGroup)
+  omopgenerics::assertLogical(sex, length = 1)
+
+  result <- purrr::map(omopTableName,
+                       function(x) {
+                         checkOmopTable(cdm[[x]])
+                         if(cdm[[x]] |> dplyr::tally() |> dplyr::pull("n") == 0) {
+                           cli::cli_warn(paste0(x, " omop table is empty. Returning an empty summarised omop table."))
+                           return(omopgenerics::emptySummarisedResult())
+                         }
+                         summariseRecordCountInternal(x,
+                                                      cdm = cdm,
+                                                      unit = unit,
+                                                      unitInterval = unitInterval,
+                                                      ageGroup = ageGroup,
+                                                     sex = sex)
+                       }
+  ) |>
+    dplyr::bind_rows()
```

Comment:
can you use omopgenerics::bind()? thats what we use to bind multiple summarised_results

Reply:
I dont know if it is needed, we can discuss in sprint meeting